src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -0,0 +1,84 @@
+import { AfterViewChecked, Component, ElementRef, Input, OnChanges, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { ItemData } from '../../services/item-datasource.service';
+import { interval, merge, Observable } from 'rxjs';
+import { filter, map, startWith, switchMap } from 'rxjs/operators';
+import { SECONDS } from 'src/app/shared/helpers/duration';
+import { isNotUndefined } from 'src/app/shared/helpers/null-undefined-predicates';
+import { ItemTaskService } from '../../services/item-task.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { capitalize } from 'src/app/shared/helpers/case_conversion';
+import { ItemTaskInitService } from '../../services/item-task-init.service';
+import { ItemTaskAnswerService } from '../../services/item-task-answer.service';
+import { ItemTaskViewsService } from '../../services/item-task-views.service';
+
+const initialHeight = 1200;
+const heightSyncInterval = 0.2*SECONDS;
+
+interface TaskTab {
+  name: string,
+  view: string,
+}
+
+@Component({
+  selector: 'alg-item-display',
+  templateUrl: './item-display.component.html',
+  styleUrls: [ './item-display.component.scss' ],
+  providers: [ ItemTaskService, ItemTaskInitService, ItemTaskAnswerService, ItemTaskViewsService ],
+})
+export class ItemDisplayComponent implements OnInit, AfterViewChecked, OnChanges {
+  @Input() itemData?: ItemData;
+  @ViewChild('iframe') iframe?: ElementRef<HTMLIFrameElement>;
+
+  state$ = this.taskService.task$.pipe(mapToFetchState());
+
+  tabs$: Observable<TaskTab[]> = this.taskService.views$.pipe(
+    map(views => views.map(view => ({ view, name: this.getTabNameByView(view) }))),
+  );
+  activeTabView$ = this.taskService.activeView$;
+
+  iframeSrc$ = this.taskService.iframeSrc$;
+
+  // Start updating the iframe height to match the task's height
+  iframeHeight$ = merge(
+    this.taskService.task$.pipe(switchMap(task => interval(heightSyncInterval).pipe(switchMap(() => task.getHeight())))),
+    this.taskService.display$.pipe(map(({ height }) => height), filter(isNotUndefined)),
+  ).pipe(startWith(initialHeight), map(height => height + 40));
```

Comment:
40 would be worth a constant, I don't even know what it stands for.

Reply:
Fixed